### PR TITLE
Process DMs and add generate message function

### PR DESCRIPTION
**Pull request by Dom-TC**

### CODE REVIEW

File: ledger_bot/message_generator.py [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+def generate_transaction_status_message(
+    seller: discord.Member,
+    buyer: discord.Member,
+    wine_name: str,
+    wine_price: float,
+    config: dict,
+    is_update: Optional[bool] = False,
+    is_approved: Optional[bool] = False,
+    is_buyer_paid: Optional[bool] = False,
+    is_seller_paid: Optional[bool] = False,
```
Comment (OllyDS) [inline ledger_bot/message_generator.py]: Small nitpick but I was wondering if we should rename this var to something more descriptive like `is_payment_confirmed`. Likewise for `is_buyer_delivered` - could be `is_delivery_confirmed`?

File: ledger_bot/slash_commands.py [hunk 2]
```diff
@@ -59,6 +60,9 @@ async def list_transaction(
         price: float,
     ):
         """Add transaction to Airtable."""
+        # Discord Interactions need to be responded to in <3s or they time out.  We take longer, so defer the interaction.
```
Comment (OllyDS) [inline ledger_bot/slash_commands.py:63]: Ah nice! This may be why Mark and I got errors despite the table being updated.

**Review by OllyDS — APPROVED** [review]

File: ledger_bot/message_generator.py (hunk 1 above)
Comment (Dom-TC) [reply ledger_bot/message_generator.py]: I thought of that, but wanted to keep it clear whether it's buyer vs seller.  Would `is_marked_paid_by_buyer` etc. be better?

File: ledger_bot/slash_commands.py (hunk 2 above)
Comment (Dom-TC) [reply ledger_bot/slash_commands.py:63]: Yeah, I only discovered this late last night!  I don't know why it was fine when I was doing it the day before, but I guess we must have been right on the edge

File: ledger_bot/message_generator.py (hunk 1 above)
Comment (OllyDS) [reply ledger_bot/message_generator.py]: Yeah I like that 👌

File: ledger_bot/slash_commands.py (hunk 2 above)
Comment (OllyDS) [reply ledger_bot/slash_commands.py:63]: Great catch!

**Pull request merged**